### PR TITLE
Fix pilot tests to use www.google.com instead of nghttp2.org

### PR DESCRIPTION
**Pull request by nmittler**
nghttp2.org is currently unreachable.

### CODE REVIEW

**Comment by kyessenov** [comment]
The test checks some headers I think. Maybe use istio.io at least?

**Comment by ldemailly** [comment]
I use istio.io too for some fortio tests; for external secure grpc I use fortio.istio.io but that's also because if it breaks I then have a big incentive to fix it/find out why that would "flake"

**Comment by nmittler** [comment]
@ldemailly @kyessenov I think I'll stick with google.com, since there is already a config for it.

File: tests/e2e/tests/pilot/testdata/v1alpha1/egress-rule-nghttp2.yaml.tmpl [hunk 1]
```diff
@@ -1,11 +0,0 @@
-apiVersion: config.istio.io/v1alpha2
```
Comment (ldemailly) [inline tests/e2e/tests/pilot/testdata/v1alpha1/egress-rule-nghttp2.yaml.tmpl]: don't you need a replacement rule ? there is already one for google?

Comment (nmittler) [reply tests/e2e/tests/pilot/testdata/v1alpha1/egress-rule-nghttp2.yaml.tmpl]: https://github.com/istio/istio/blob/master/tests/e2e/tests/pilot/testdata/v1alpha1/egress-rule-google.yaml.tmpl

**Comment by rshriram** [comment]
the only problem is that google.com is very forgiving in terms of headers/fields in the request, and has lots of features to auto fallback to h1. It would be better to pick a site that specifically has h2 (that was the rationale behind nghttp2.org)

**Pull request merged**